DirectInput/Joystick/joystick.cpp

```diff
@@ -256,169 +239,100 @@ HRESULT InitDirectInput( HWND hDlg )
                                                ( VOID* )hDlg, DIDFT_ALL ) ) )
         return hr;
 
+    FillJoystickInfo( hDlg );
+
     return S_OK;
 }
 
 
 //-----------------------------------------------------------------------------
-// Enum each PNP device using WMI and check each device ID to see if it contains 
-// "IG_" (ex. "VID_045E&PID_028E&IG_00").  If it does, then it's an XInput device
-// Unfortunately this information can not be found by just using DirectInput.
-// Checking against a VID/PID of 0x028E/0x045E won't find 3rd party or future 
-// XInput devices.
-//
-// This function stores the list of xinput devices in a linked list 
-// at g_pXInputDeviceList, and IsXInputDevice() searchs that linked list
+// Returns true if the DirectInput device is also an XInput device.
+// Checks if device ID contains "IG_" (ex. "\\?\HID#VID_045E&PID_02A1&IG_00").
+// If it does, then it's an XInput device.
 //-----------------------------------------------------------------------------
-HRESULT SetupForIsXInputDevice()
+bool IsXInputDevice( LPDIRECTINPUTDEVICE8 pJoystick )
 {
-    IWbemServices* pIWbemServices = nullptr;
-    IEnumWbemClassObject* pEnumDevices = nullptr;
-    IWbemLocator* pIWbemLocator = nullptr;
-    IWbemClassObject* pDevices[20] = {0};
-    BSTR bstrDeviceID = nullptr;
-    BSTR bstrClassName = nullptr;
-    BSTR bstrNamespace = nullptr;
-    DWORD uReturned = 0;
-    bool bCleanupCOM = false;
-    UINT iDevice = 0;
-    VARIANT var;
-    HRESULT hr;
+    DIPROPGUIDANDPATH dip;

```

Comment:
Using DI directly instead of WMI does look a lot simpler. Have you been able to check this on more than one version of Windows?

Reply:
Unfortunately I only have Win10. Will try to install win 7 vm for testing this.

Reply:
Same approach actually used in Google Chrome: https://chromium.googlesource.com/chromium/chromium/+/trunk/content/browser/gamepad/raw_input_data_fetcher_win.cc#233